osmo_jupyter/dataset.py

```diff
@@ -0,0 +1,267 @@
+    images = pivot["image"][experiment_df["ROI"].values[0]]
+    pivot.drop("image", axis=1, inplace=True)
+    # Flatten the pivot table index, and add images back in
+    pivot.columns = [" ".join(col[::-1]).strip() for col in pivot.columns.values]
+    pivot["image"] = images
+
+    return pivot
+
+
+def open_and_combine_process_experiment_results(
+    process_experiment_result_filepaths: List[str],
+    ROI_names: List[str] = None,
+    msorm_types: List[str] = ["r_msorm", "g_msorm", "b_msorm"],
+) -> pd.DataFrame:
+    """
+        Open multiple process experiment result files and combine into a single DataFrame with one row per image.
+
+        Args:
+            process_experiment_result_filepaths: A list of filepaths to process experiment summary statistics files.
+            ROI_names: Optional. A list of ROI names to select. Defaults to all ROIs present.
+            msorm_types: Optional. A list of MSORM column names to select. Defaults to all RGB channel MSORMs.
+        Returns:
+            DataFrame of all summary statistics flattened to one row per image with all selected ROIs.
+    """
+    all_roi_data = pd.concat(
+        [
+            pd.read_csv(results_filepath, parse_dates=["timestamp"])
+            for results_filepath in process_experiment_result_filepaths
+        ]
+    )
+
+    return pivot_process_experiment_results_on_ROI(all_roi_data, ROI_names, msorm_types)
+
+
+def get_all_experiment_images(
+    local_sync_directory: str, experiment_names: List[str]
+) -> pd.DataFrame:
+    """
+        Get a DataFrame of all image files across multiple experiment data directories.
+
+        Args:
+            local_sync_directory: The local data directory, usually ~/osmo/cosmobot-data-sets
+            experiment_names: A list of experiment directory names in the local sync directory.
+        Returns:
+            DataFrame of all image file names and the corresponding experiment name.
+    """
+    all_images = pd.concat(
+        [
+            pd.DataFrame(
+                {
+                    "experiment": experiment_name,
+                    "image": os.listdir(
+                        os.path.join(local_sync_directory, experiment_name)
+                    ),
+                },
+                dtype="object",  # Ensure correct dtype when no images are found
+            )
+            for experiment_name in experiment_names
+        ]
+    )
+
+    # Filter out experiment log files
+    return all_images.drop(all_images[~all_images["image"].str.contains("jpeg")].index)
+
+
+def filter_equilibrated_images(equilibration_range: pd.Series, df: pd.DataFrame):
+    """
+        Filter a datetime indexed DataFrame to a given equilibration range
+
+        Args:
+            equilibration_range: A Series with indexed start_time and end_time values
+            df: A datetime indexed DataFrame
+        Returns:
+            A fitlered df containing only values whose index is between the given start_time and end_tim.
```

Comment:
nit. typing a little fast now, are we
```suggestion
            A filtered df containing only values whose index is between the given start_time and end_time.
```